### PR TITLE
[gunicorn] Adding environment configuration

### PR DESCRIPTION
**Pull request by john-bodley**
Description of changeset: 

This PR mimics logic [here](https://github.com/benoitc/gunicorn/blob/master/gunicorn/app/base.py#L150) to allow additional Gunicorn configuration to be defined via the `GUNICORN_CMD_ARGS` environment variable, i.e., 
```
GUNICORN_CMD_ARGS='--statsd-host=localhost:8125' knowledge_repo deploy
```

Test Plan: 

Verified that the command line args where present.

Auto-reviewers: @NiharikaRay @matthewwardrop @earthmancash @danfrankj

### CODE REVIEW

File: knowledge_repo/app/deploy/gunicorn.py [hunk 1]
```diff
@@ -21,12 +21,21 @@ def __init__(self, *args, **kwargs):
...
+        # Lastly, update the configuration with any command line settings.
+        args = {
             'bind': u'{}:{}'.format(self.host, self.port),
             'workers': self.workers,
-            'timeout': self.timeout
+            'timeout': self.timeout,
         }
-        for key, value in options.items():
+
+        for key, value in args.items():
```
Comment (NiharikaRay) [inline knowledge_repo/app/deploy/gunicorn.py]: do you still need this iteration over `args.items()`, when you're doing the same thing above on line 26?

File: knowledge_repo/app/deploy/gunicorn.py [hunk 2]
```diff
@@ -21,12 +21,21 @@ def __init__(self, *args, **kwargs):
         BaseApplication.__init__(self)
 
     def load_config(self):
-        options = {
+        args = self.cfg.parser().parse_args(self.cfg.get_cmd_args_from_env())
+
+        # Load up environment configuration.
+        for key, value in vars(args).items():
+            if key != 'args' and value is not None:
```
Comment (NiharikaRay) [inline knowledge_repo/app/deploy/gunicorn.py]: why are we filtering out for keys that are `args` here?

File: knowledge_repo/app/deploy/gunicorn.py (hunk 1 above)
Comment (john-bodley) [reply knowledge_repo/app/deploy/gunicorn.py]: The args here are different. These are the args from the `GUNICORN_CMD_ARGS ` environment variable.

File: knowledge_repo/app/deploy/gunicorn.py (hunk 2 above)
Comment (john-bodley) [reply knowledge_repo/app/deploy/gunicorn.py]: `args` here is a config object which contains the `arg` field which should be ignore. This logic is from [here](https://github.com/benoitc/gunicorn/blob/master/gunicorn/app/base.py#L150).

File: knowledge_repo/app/deploy/gunicorn.py [hunk 3]
```diff
@@ -21,12 +21,21 @@ def __init__(self, *args, **kwargs):
...
+                self.cfg.set(key, value)
+
+        # Lastly, update the configuration with any command line settings.
+        args = {
```
Comment (danfrankj) [inline knowledge_repo/app/deploy/gunicorn.py]: @john-bodley for readability and future-proofing, let's have only use `args` once. e.g. above `env_args` and here `command_line_args`

Comment (john-bodley) [reply knowledge_repo/app/deploy/gunicorn.py]: 👍

**Comment by john-bodley** [comment]
@danfrankj I addressed your comment.

**Pull request merged**